AzureStack/2015-04-01/swagger/GalleryAdmin.json

```diff
@@ -0,0 +1,502 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "GalleryAdminClient",
+    "description": "Management client for azurestack gallery admin operations.",
+    "version": "2015-04-01"
+  },
+  "host": "publicapi.azurestack.local",
+  "schemes": [
+    "https"
+  ],
+    "consumes": [
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "security": [
+    {
+      "azure_auth": [
+        "user_impersonation"
+      ]
+    }
+  ],
+  "securityDefinitions": {
+    "azure_auth": {
+      "type": "oauth2",
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "flow": "implicit",
+      "description": "Azure Active Directory OAuth2 Flow",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "paths": {
+      "/subscriptions/{subscriptionId}/providers/microsoft.gallery.admin/galleryItems": {
+      "get": {
+        "tags": [
+          "AdminGalleryItems"
+        ],
+        "description": "Lists gallery registrations.",
+        "operationId": "AdminGalleryItems_List",
+        "consumes": [],
+        "produces": [
+          "application/json",
+          "text/json"
+        ],
+        "parameters": [
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "type": "array",
+              "items": {
+                "$ref": "#/definitions/ResourceDefinition"
+              }
+            }
+          }
+        }
+      },
+      "post": {
+        "tags": [
+          "AdminGalleryItems"
+        ],
+        "description": "Uploads a provider gallery item to the storage.",
+        "operationId": "AdminGalleryItems_Upload",
+        "consumes": [         
+          "application/json",
+          "text/json"
+          ],
+        "produces": [
+          "application/json",
+          "text/json"
+        ],
+        "parameters": [
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "name": "galleryItemUri",
+            "in": "body",
+            "description": "Gallery Item Uri from where the gallery package is downloaded.",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/GalleryItemUriPayload"
+            }
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/ResourceDefinition"
+            }
+          },
+          "201": {
```

Comment:
- is this a long running operation?
- Can you explain in the description, when will the server return a 200 and when will it return a 201?